### ISSUE TITLE
Multiple issues with template literal type narrowing and relations

### ISSUE THREAD

**Issue by ahejlsberg** [microsoft/TypeScript#46125]
### 🕗 Version & Regression Information

Every version.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMAjACgB4C5EANhxwC0AJAN4DOUATjGAOYC+uANIgJ467lW0ONEARgENquAJSIyAKERzEMYIkyIAvOo6SycgPQ7EAdWoJ6AG3aJqAUwAOcalAqIr1Y9URQAFsKgf2NqycAEwQAcl84ADcXU2EbWXlGaSTpUEhYBGQAJkwcXjp6Nk48AmIeGgLmRAAfRFCRalCtBLlFZQw1DXZm+V7EDABuXX0AOVFjAHcrII84OobQtgpPOBBTGcErEsJSSgr+XBr50VCWxCSUtOh4JGAAZlzEfIYirlLd56YJKTO2lXVVHV3lBAlAmj8+vJBsNEGNXHApjMoHNQsDQadeklen8LACgTsQVRwTJIXJ2ENEHpYXBfGBxgjpmcLtIgA)

### 💻 Code

```ts
function f1(x: `foo-${string}`, y: `${string}-bar`) {
    if (x === y) {  // Wrongly reports error that types don't overlap (they do, e.g. 'foo-xxx-bar')
    }
}

function f2(x: string, y: `foo-${string}` | 'bar') {
    if (x === y) {
        x;  // Narrowed to 'bar', should be `foo-${string}` | 'bar'
    }
}

function f3(x: string, y: `foo-${string}`) {
    if (x === 'foo-test') {
        x;  // Narrowed to 'foo-test'
    }
    if (y === 'foo-test') {
        y;  // Not narrowed but should be
    }
}
```

### 🙁 Actual behavior

See above.

### 🙂 Expected behavior

See above.

**Comment by craigphicks** [comment]
That was very kind of you to attempt to calssify fixed prefixes - and fixed mid?/post-fixes too, all in the first release.  I would have thought that classifying fixed prefixes alone would be sufficient, and not necessarily in the first release.

How about posting behavior/usage specifications and a list of tentatively planned vs released features along with the pull?  A bird in the hand is worth two in the bush.  Again thank you for all your efforts and for this great feature.
